[DOCS][101] Update first quick start with mappings examples

leemthompo opened the pull request:
# 📸[URL PREVIEW](https://elasticsearch_bk_113558.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/getting-started.html)


Originally envisioned this being standalone companion to basics page about [indices, documents, mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/documents-indices.html) but we discussed that might actually fit right into the first quick start?

My only concern is this is a little long now 🤔 

TODO:

- Align on titles for these and heading abbreviations etc.

leemthompo committed:
[DOCS] Update first quick start with mappings examples

leemthompo committed:
Fix test?

leemthompo committed:
Fix tests, update verbiage

Added my-explicit-mappings-books to TESTSETUP and TEARDOWN.
Skipped the initial PUT /books test as the index is already set up.
Modified the GET /_cat/indices command to include specific parameters.
Changed a PUT request to POST for adding a document.
Updated the description of the dynamically added field's data type from keyword to text.
Made minor wording adjustments throughout the document.

leemthompo committed:
Fix heading level

leemthompo committed:
Use verbs for heading consistency

leemthompo committed:
Remove superfluous list item

leemthompo committed:
I guess need some context right

leemthompo committed:
Fix typo

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,81 @@
+<titleabbrev>Basics: Indices, documents, and mappings</titleabbrev>
 ++++
 
 In this quick start guide, you'll learn how to do the following tasks:
```
One element I think we should ensure we talk about is pre-reqs.. in case this person hit this article they may not realize they need to have an environment to use. We don't need to put the install instructions in here but at least a reference on where to go to setup their environment.

leemthompo replied:
Added in https://github.com/elastic/elasticsearch/pull/113558/commits/31d8006392b2eb3e578309b43493ad420c6a99e6

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,81 @@
+
+[source,console]
+----
+GET /_cat/indices?v&h=index,docs.count&s=index:asc
```
Do we want to show what the response looks like for this. You have example responses for other examples put not this one. Just thinking out loud.

leemthompo replied:
I'll resolve this because we talk about examples in general in another https://github.com/elastic/elasticsearch/pull/113558#discussion_r1778178420

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +222,100 @@ You should receive a response indicating there were no errors.
+
+[source,console]
+----
+GET /books/_mapping
```
Your previous step says "see how dynamic mapping works" but you dont show a result. This again may be an opportunity to show a result as to what happened?

leemthompo replied:
resolve this because we talk about examples in general in another [comment](https://github.com/elastic/elasticsearch/pull/113558#discussion_r1778178420)

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +222,100 @@ You should receive a response indicating there were no errors.
 
 [discrete]
-[[search-all-documents]]
+[[getting-started-search-all-documents]]
```
again don't want to litter everything with responses but should the search data examples show what to expect since they are get commands?

leemthompo replied:
great question and I'm not sure because we would have to collapse them to make the page readable, so will need to find consensus after feedback

it might be easier to not print responses and put the onus on the user to actually run the examples which is the whole point...but it is a balancing act

shainaraskas replied:
I would like collapsible results here, or truncated ones

leemthompo replied:
Due to high demand, I've gone all-in on responses and annotations

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -232,4 +330,19 @@ GET books/_search
+----
+// TEST[skip:handled by setup/teardown]
+
+
```
I don't know how this works but assume the "setup elasticsearch" at the bottom right of the article is there because of the next node in the ToC?

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,81 @@
+<titleabbrev>Basics: Indices, documents, and mappings</titleabbrev>
 ++++
 
 In this quick start guide, you'll learn how to do the following tasks:
```
Quickstart articles like this should still have an introductory paragraph. Introductory paragraphs can help search engines when searching for content.

leemthompo replied:
Roger that!

leemthompo replied:
because the intro paragraph will be prose version of the "in this quick start you'll learn how to do x,y,z" list, wonder if should remove that list entirely and let the "**On this page**" serve that porpoise 🐬

leemthompo replied:
add intro in https://github.com/elastic/elasticsearch/pull/113558/commits/31d8006392b2eb3e578309b43493ad420c6a99e6

georgewallace commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,81 @@
-Submit the following indexing request to add a single document to the
-`books` index.
-The request automatically creates the index.
+* Create an index
```
should these be bookmark links to the section, Google can use lists like this to build cards on search that can help provide a card like experience that helps customers navigate

leemthompo replied:
can be totally, we have the right-side "On this page" that is automatically generated for us, so I'm not sure what the best practice is here

will be great to establish pattern for future in any case once we decide 👍

leemthompo replied:
replaced with prose in https://github.com/elastic/elasticsearch/pull/113558/commits/31d8006392b2eb3e578309b43493ad420c6a99e6

leemthompo committed:
Expand intro, add prerequisites, add hints for Console usage

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
 [[getting-started]]
-== Quick start: Add data using Elasticsearch APIs
+== Quick start: Indices, documents, and mappings
```
this title doesn't encompass the "search" chapter of the tutorial. consider "Index and search documents" or something that gets search in there.

mappings is kind of a nuance of indexing

leemthompo replied:
💯 I have a real brain block on how to title these

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
 // TEST[continued]
 
-The `_source` of each hit contains the original
+The `_source` metadata field of each hit contains the original
```
link off to the metadata fields docs?

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -232,4 +339,19 @@ GET books/_search
+[[getting-started-delete-indices]]
+=== Delete index
+
+Delete indices using the <<indices-delete-index,`DELETE` index API>>.
```
why would you want to delete an index? might want a use case here

leemthompo replied:
not critical but feels helpful and like a natural bookend in the spirit of CRUD

The main use case for the tutorial reader is to start from scratch if an index ever gets funky

I'll add an _optional_ tag.

Happy to delete this if it's overkill tho.

shainaraskas reviewed (commented):
provided mostly tutorial style and depth comments inline. regarding your question on direction: I like the idea of a quick "index a doc and see a result" type tutorial to get people familiar with playing with the API, but I wonder if, in a better IA world, this would be a top-level tutorial in an index series, that we could just link to prominently. the thought in the back of my mind is: we need tutorials in the index content, so what would go there, if not this style of tutorial? 

given that we don't live in said world, I'm cool with this getting published

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
-* Run basic searches.
+[TIP]
+====
+By default the examples are in Dev Tools Console syntax, but you can {kibana-ref}/console-kibana.html#import-export-console-requests[export to other programming languages] in the Console UI.
```
- The docs examples should be translated into other PL's in our docs code widget, so 'only available' isn't accurate, I just wasn't saying straight up "also in Python, Ruby, and JS" because that conversion process isn't immediate and doesn't have 100% coverage yet. So I don't want to say "sometimes available in those languages too".

- Also:
  - They will always be available in at least Console + curl 
  - The widget does display them in Console _by default_

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
 
+[discrete]
+[[getting-started-index-creation]]
+=== Create index
```
> Should we consider numbering them because they're a procedure?

yes!

> rabbit hole
let's leave this particular rabbit family alone

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
+
+[discrete]
+[[getting-started-add-documents]]
+=== Add data
```
yes x 1000000

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
 ++++
 
-In this quick start guide, you'll learn how to do the following tasks:
+This quick start guide is a hands-on introduction to the fundamental concepts of Elasticsearch: indices, documents, and field type mappings. 
```
👍 x 100000

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,51 +1,90 @@
+
+[TIP]
+====
+If the index didn't already exist, the request would automatically create it.
```
yes x 1

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -72,12 +111,11 @@ The response includes metadata that {es} generates for the document including a
-must be newline-delimited JSON (NDJSON). Each line must end in a newline
-character (`\n`), including the last line.
+Use the <<docs-bulk,`_bulk` endpoint>> to add multiple documents in one request. Bulk data
+must be newline-delimited JSON (NDJSON).
```
yes 🔍

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
+[[getting-started-mappings-and-data-types]]
+=== Define mappings and data types
+
+<<elasticsearch-intro-documents-fields-mappings,Mappings>> define how data is stored and indexed in {es}, much like a  schema for a relational database.
```
what's a garden without flowers?

...oh wait this isn't a garden...

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
+When using dynamic mapping, {es} automatically creates mappings for new fields by default.
+The documents we've added so far have used dynamic mapping, because we didn't specify a mapping when creating the index.
+
+Let's add a new document to the `books` index with a field that doesn't exist, to see how dynamic mapping works.
```
"let's" drop it then

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
+  "author": "F. Scott Fitzgerald",
+  "release_date": "1925-04-10",
+  "page_count": 180,
+  "new_field": "This is a dynamically added field"
```
yup need to be consistent with annotations and responses 👍

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
+[[getting-started-explicit-mapping]]
+==== Define explicit mapping
+
+Create an index named `my-explicit-mappings-books` with explicit mappings:
```
yes indeedy additional annotations incoming

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +231,100 @@ You should receive a response indicating there were no errors.
+[[getting-started-combined-mapping]]
+==== Combine dynamic and explicit mappings
+
+When an index has the `dynamic` flag set to `true`, you can add new fields to documents without updating the mapping.
```
surely and I've gone a step further and updated the main mappings page and linked there too with some hopefully helpful info about combining both types

leemthompo committed:
Update mappings page with combo info

leemthompo committed:
Updates per feedback

- Updated title and titleabbrev to focus on indexing and searching using APIs
- Added links
- Clarified explanation about code examples and language conversion
- Added numbered steps (1-5)
- Added example responses
- Included more detailed explanations with numbered annotations in code snippets
- Added tip about other ways to add data to Elasticsearch
- Added more context and explanations for search operations

leemthompo commented:
Thanks `@shainaraskas`. Not pinging you on a holiday day. :)

I've made updates in https://github.com/elastic/elasticsearch/pull/113558/commits/3a3a9a47ef0606ad91544f4d13c4c8d6b6cb7b7e. See commit message for summary.

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
-Indexed documents are available for search in near real-time.
+Explicit mappings are defined at index creation, and documents must conform to these mappings.
+When an index has the `dynamic` flag set to `true`, you can add new fields to documents without updating the mapping.
+This allows you to <<mapping-combine-explicit-dynamic,combine explicit and dynamic mappings>>.
```
added in https://github.com/elastic/elasticsearch/pull/113558/commits/c0eba9fcafe7b01d0321ce6a732782752b269b33

leemthompo commented on docs/reference/quickstart/index.asciidoc:
```diff
@@ -15,7 +15,7 @@ Get started <<run-elasticsearch-locally,locally in Docker>> , or see our <<elast
 == Hands-on quick starts
 
-* <<getting-started,Basics: Add data using APIs>>. Learn how to add data to {es} and perform basic searches.
+* <<getting-started,Basics: Indices, documents, and mappings>>. Learn about indices, documents, and mappings, and perform a basic search.
```
```suggestion
* <<getting-started,Basics: Index and search data using {es} APIs>>. Learn about indices, documents, and mappings, and perform a basic search.
```

leemthompo committed:
Update link text

leemthompo commented:
🤔 Because we're doing a lot of explanation here, with response annotations and links to every concept/API call, we're already in _tutorial_ rather than _quick start_ territory. Perhaps tutorials is a better umbrella. These are a lot more pedagogical than "copy these few commands and be on your way".

leemthompo committed:
Minor touchups, mainly to rerun CI

leemthompo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'main' into qs-basics

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
 [discrete]
-[[qs-match-query]]
+[[getting-started-match-query]]
 ==== `match` query
```
🤔 @georgewallace @shainaraskas @marciw  I'm wondering if we should even remove the match query here and make it the first query in the full-text search tutorial.

That way in this first tutorial we just present the `GET books/_search` call, as super basic "R" in "CRUD" and say "Learn more about search queries in foo tutorial". 🤷

leemthompo replied:
we'll revisit this line of thought once the next one is out the gate

shainaraskas commented on docs/reference/mapping.asciidoc:
```diff
@@ -75,9 +75,23 @@ reindexing. You can use runtime fields in conjunction with indexed fields to
+
+You can _update_ mappings for existing fields under certain conditions:
+
+. You can add new fields to an existing mapping.
+. You can change the mapping for fields that don't contain any data.
+. You can add new <<multi-fields,multi-fields>> for existing fields.
```
this should be a UL, not an OL

```suggestion
* You can add new fields to an existing mapping.
* You can change the mapping for fields that don't contain any data.
* You can add new <<multi-fields,multi-fields>> for existing fields.
```

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
+
+A `200` response indicates the index was created successfully.
+
+.Expand to see example response
```
:2c:  the arrow with a simpler label is more concise

```suggestion
.Example response
```

leemthompo replied:
agreeeeeee

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
+----
+// TEST[skip: index already setup]
+
+A `200` response indicates the index was created successfully.
```
wonder if the console tester would blow up if you put the http code inline ... ex below

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
+===============
+[source,console-result]
+----
+{
```
```suggestion
HTTP 200
{
```

leemthompo replied:
gonna elide this one

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
+The response includes metadata that {es} generates for the document, including a unique `_id` for the document within the index.
 
-.Expand to see example response
+.Expand to see example response with annotations
```
the "w/ annotations" I think is unnecessary here

leemthompo replied:
💯

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
 ----
-// TEST[skip:TODO]
+// TEST[s/O0lG2IsBaSa7VYx_rEia/*/]
+<1> The `_index` field indicates the index the document was added to.
```
this is a lot. do we have this anywhere else? perhaps we defer to the metadata docs for this level of detail.

leemthompo replied:
it's optional tho so I'm wondering if we leave it in because if you don't know you don't know and it's helpful

leemthompo replied:
it is a lot but don't see resource where everything is laid out together

see https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-fields.html


let me know how strongly feel about this one

shainaraskas replied:
are we possibly missing metadata fields in the linked reference then?  I agree with keeping annotations on the non-metadata fields

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
+When using dynamic mapping, {es} automatically creates mappings for new fields by default.
+The documents we've added so far have used dynamic mapping, because we didn't specify a mapping when creating the index.
+
+Add a new document to the `books` index with a field that doesn't exist, to see how dynamic mapping works.
```
```suggestion
To see how dynamic mapping works, add a new document to the `books` index with a field that doesn't exist.
```

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
+----
+// TEST[continued]
+<1> This field is dynamically added to the document.
+View the mapping for the `books` index. The new field `new_field` has been added to the mapping with a `text` data type.
```
```suggestion

View the mapping for the `books` index. The new field `new_field` has been added to the mapping with a `text` data type.
```

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
+  "author": "F. Scott Fitzgerald",
+  "release_date": "1925-04-10",
+  "page_count": 180,
+  "new_field": "I'm a new field added dynamically" <1>
```
in my original note, I meant we could fancy it up a little to make it feel more like a book field. if you don't feel like changing the example field, then I don't think the annotation is necessary.

also: I say this in another comment, but is the field added dynamically? 

```suggestion
  "language": "EN" <1>
```

leemthompo replied:
ah yes thanks

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
+}
+----
+// TEST[continued]
+<1> This field is dynamically added to the document.
```
is it added dynamically to the document, or is it added dynamically to the index mapping? 

alt:

```suggestion
<1> The new field.
```

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -1,83 +1,145 @@
 [source,console]
 ----
 POST books/_doc
 {"name": "Snow Crash", "author": "Neal Stephenson", "release_date": "1992-06-01", "page_count": 470}
```
would be nice to see this one nicely formatted like the great gatsby below

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
+  }
+}
+----
+// TEST[continued]
+<1> Disables dynamic mapping for the index. Documents containing fields not defined in the mapping will be rejected.
+<2> The `properties` object defines the fields and their data types for documents in this index.
+<3> Defines the "name" field as type "text", suitable for full-text search on book titles.
+<4> Specifies the "release_date" field as a "date" type with the specific format `yyyy-MM-dd`.
+<5> Sets the "page_count" field as an "integer" type for storing the number of pages.
```
again possibly overkill. what I was looking for was something like 

> Pass each field's properties as a JSON object. This object should contain the field type and any additional [mapping parameters](https://elasticsearch_bk_113558.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/mapping-params.html).

would skip annotations 3-5 in favor of this.

leemthompo replied:
> Pass each field's properties as a JSON object. This object should contain the field type and any additional [mapping parameters](https://elasticsearch_bk_113558.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/mapping-params.html).


Ooooh nice

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
 
-Indexed documents are available for search in near real-time.
+Explicit mappings are defined at index creation, and documents must conform to these mappings.
+When an index has the `dynamic` flag set to `true`, you can add new fields to documents without updating the mapping.
```
can you add additional manual mappings at any time? this is not captured on this page if so

leemthompo replied:
update mappings.asciidoc to be clearer here 👍

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +255,216 @@ You should receive a response indicating there were no errors.
-This is the standard query for performing full-text search, including fuzzy matching and phrase searches.
+This is the standard query for full-text searches.
 
 Run the following command to search the `books` index for documents containing `brave` in the `name` field:
```
nit: we didn't get them to add this book in the tutorial.

leemthompo replied:
huh? Brave New World? What am I missing?

shainaraskas replied:
if someone is following this tutorial from top to bottom, they'd currently only have two books in their index: snow crash and the great gatsby

so the search would return no results

leemthompo replied:
What about the bulk api call? You've added multiple additional documents there

leemthompo replied:
https://elasticsearch_bk_113558.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/getting-started.html#getting-started-add-multiple-documents

shainaraskas replied:
hi im illiterate this pr is perfect :hide:

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -232,4 +479,67 @@ GET books/_search
+=== Step 5: Delete your indices (optional)
+
+When following along with examples, you might want to delete an index to start from scratch.
+You can delete indices using the <<indices-delete-index,`DELETE` index API>>.
```
```suggestion
You can delete indices using the <<indices-delete-index,Delete index API>>.
```

leemthompo committed:
Apply suggestions from code review

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>

leemthompo committed:
Further edits per review

leemthompo committed:
Reword mappings update, add clarity

leemthompo committed:
Tweak mapping updates wording

leemthompo committed:
Really clarify this time

leemthompo committed:
Edit update verbiage

leemthompo committed:
Typo

kderusso reviewed (commented):
I added some clarifications on mapping behavior that we should probably address.

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +260,215 @@ You should receive a response indicating there were no errors.
+          "page_count": 268
+        }
+      },
+      ... (truncated)
```
haha me neither I think I forgot to change continued to skip here but it seems to work 😄

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -232,4 +483,65 @@ GET books/_search
+DELETE /books
+DELETE /my-explicit-mappings-books
+----
+// TEST[skip:handled by setup/teardown]
```
docscheck is green

kderusso replied:
Right, I mean if you didn't skip this test

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +260,215 @@ You should receive a response indicating there were no errors.
+  "index": "my-explicit-mappings-books"
+}
+----
+// TEST[skip:already created in setup]
```
I think this is a typo 🔍

leemthompo replied:
Ah right, yeah I'm still a noobie with these tests TBH, the syntax and structure doesn't sit logically with me :)

leemthompo committed:
Apply Kathleen's suggestions

Co-authored-by: Kathleen DeRusso <kathleen.derusso@elastic.co>

leemthompo committed:
Clarify multi-field mapping updates

leemthompo commented on docs/reference/mapping.asciidoc:
```diff
@@ -75,9 +75,26 @@ reindexing. You can use runtime fields in conjunction with indexed fields to
+
+* You can add new fields to an existing mapping at any time, explicitly or dynamically.
+* You can change the mapping for fields that don't contain any data.
+* You can add new <<multi-fields,multi-fields>> for existing fields.
```
addressed in https://github.com/elastic/elasticsearch/pull/113558/commits/11a38d0913e102cd525e55cbd46c9393ca4a8514

leemthompo committed:
Link to update mapping API

leemthompo committed:
Mention some mapping params can be updated

leemthompo committed:
Further clarifications

leemthompo commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -193,31 +260,215 @@ You should receive a response indicating there were no errors.
+[[getting-started-combined-mapping]]
+==== Combine dynamic and explicit mappings
+
+Explicit mappings are defined at index creation, and documents must conform to these mappings.
```
added

shainaraskas commented on docs/reference/quickstart/getting-started.asciidoc:
```diff
@@ -344,7 +350,8 @@ GET /books/_mapping
 
-Create an index named `my-explicit-mappings-books` with explicit mappings:
+Create an index named `my-explicit-mappings-books` with explicit mappings.
+Pass each field's properties as a JSON object. This object should contain the field type and any additional <<mapping-params,mapping parameters>>.
```
🤔 do we want a link to available types here?

shainaraskas reviewed (approved):
approving - ~I do see the brave new world issue as possibly blocking, but if you fix that~ you're good to go imo

leemthompo committed:
Add tweaks and links

elasticsearchmachine commented:
## 💔 Backport failed
The backport operation could not be completed due to the following error:
```
An unexpected error occurred when attempting to backport this PR.
```

You can use [sqren/backport](https://github.com/sqren/backport) to manually backport by running `backport --upstream elastic/elasticsearch --pr 113558`

leemthompo commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/113972">](https://github.com/elastic/elasticsearch/pull/113972)|
|✅|8.15|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/113973">](https://github.com/elastic/elasticsearch/pull/113973)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.3.0 -->
